compat/getaddrinfo.cc

```diff
@@ -190,7 +184,7 @@ xgetaddrinfo (const char *nodename, const char *servname,
     h_errno = 0;
 #endif
     errno = 0;
-    hp = gethostbyname(nodename);
+    const struct hostent hp = gethostbyname(nodename);
```

Comment:
Wrong type and you can actually simplify this further:

```suggestion
    const auto hp = gethostbyname(nodename);
```